source/guides/recipes/dependency-management.md

```diff
@@ -0,0 +1,101 @@
+(dependency-management-niv)=
+# Managing remote sources with niv
+
+The Nix language can be used to describe dependencies between files managed by Nix.
+Nix expressions themselves can depend on remote sources, and there are multiple ways to specify their origin, as shown in [](pinning-nixpkgs).
+
+For more automation around handling remote sources, set up [niv](https://github.com/nmattia/niv/) in your project:
+
+```shell-session
+$ nix-shell -p niv --run "niv init"
+```
+
+This command will generate `nix/sources.json` in the current directory, which is a lock file for dependencies.
+It will also create `nix/sources.nix`, which exposes those dependencies as an attribute set.
+
+:::{note}
+By default, `niv init` will add the latest revision of `nixpkgs-unstable` as a source.
+If you need the latest revision of a specific branch:
+
+```shell-session
+niv init --nixpkgs nixos/nixpkgs --nixpkgs-branch nixos-23.05
+```
+:::
+
+Import the generated `nix/sources.nix` for the top-level argument in the top-level `default.nix` and use it to refer to the Nixpkgs source directory:
+
+```nix
+{ sources ? import ./nix/sources.nix }:
+let
+  pkgs = import sources.nixpkgs {};
+  build = pkgs.hello;
+in {
+  inherit build;
+}
+```
+
+`nix-build` will call the top-level function with the default argument.
```

Comment:
Technically `nix-build` just calls it with `{}` (or the additional `--arg`/`--argstr` values).

Reply:
```suggestion
`nix-build` will call the top-level function with the empty attribute set `{}`, or with the attributes passed via [`--arg`](https://nixos.org/manual/nix/stable/command-ref/nix-build#opt-arg) or [`--argstr`](https://nixos.org/manual/nix/stable/command-ref/nix-build#opt-argstr).
```